UnityProject/Assets/Scripts/Messages/Server/AdminTools/AdminChatNotifications.cs

```diff
@@ -10,27 +10,36 @@
 /// </summary>
 public class AdminChatNotifications : ServerMessage
 {
-	public string NotificationKey;
-	public AdminChatWindow TargetWindow;
-	public int Amount;
-	public bool ClearAll;
-	public bool IsFullUpdate;
-	public string FullUpdateJson;
+	public struct AdminChatNotificationsNetMessage : NetworkMessage
+	{
+		public string NotificationKey;
+		public AdminChatWindow TargetWindow;
+		public int Amount;
+		public bool ClearAll;
+		public bool IsFullUpdate;
+		public string FullUpdateJson;
+	}
 
-	public override void Process()
+	//This is needed so the message can be discovered in NetworkManagerExtensions
+	public AdminChatNotificationsNetMessage IgnoreMe;
```

Comment:
so, couldn't you abstract this away  setting up ServerMessage<T>
When you decorate it you just have to do AdminChatNotifications : ServerMessage<AdminChatNotificationsNetMessage>
public AdminChatNotificationsNetMessage IgnoreMe;
And that gets turned into public T IgnoreMe; in ServerMessage/base message/whatever

Reply:
bit confused what you mean, that variable is there so i can link the struct to the class. I dont think you'd be able to abstract it away. I am no c# wizard, if you can give an example, i might be misunderstanding.

Reply:
yeah sorry i cant read, am tired. Dont think i could have abstracted that, i think having that empty variable is the only way to detect it for the handler

Reply:
ok cool